edge-hub/src/Microsoft.Azure.Devices.Routing.Core/Message.cs

```diff
@@ -43,11 +43,23 @@ public Message(IMessageSource messageSource, byte[] body, IDictionary<string, st
         }
 
         public Message(IMessageSource messageSource, byte[] body, IDictionary<string, string> properties, IDictionary<string, string> systemProperties, long offset, DateTime enqueuedTime, DateTime dequeuedTime)
+            : this(
+                messageSource,
+                body,
+                new ReadOnlyDictionary<string, string>(new Dictionary<string, string>(Preconditions.CheckNotNull(properties), StringComparer.OrdinalIgnoreCase)) as IReadOnlyDictionary<string, string>,
+                new ReadOnlyDictionary<string, string>(new Dictionary<string, string>(Preconditions.CheckNotNull(systemProperties), StringComparer.OrdinalIgnoreCase)),
+                offset,
+                enqueuedTime,
+                dequeuedTime)
+        {
+        }
+
+        public Message(IMessageSource messageSource, byte[] body, IReadOnlyDictionary<string, string> properties, IReadOnlyDictionary<string, string> systemProperties, long offset, DateTime enqueuedTime, DateTime dequeuedTime)
         {
             this.MessageSource = messageSource;
             this.Body = Preconditions.CheckNotNull(body);
-            this.Properties = new ReadOnlyDictionary<string, string>(new Dictionary<string, string>(Preconditions.CheckNotNull(properties), StringComparer.OrdinalIgnoreCase));
-            this.SystemProperties = new ReadOnlyDictionary<string, string>(new Dictionary<string, string>(Preconditions.CheckNotNull(systemProperties), StringComparer.OrdinalIgnoreCase));
+            this.Properties = Preconditions.CheckNotNull(properties);
+            this.SystemProperties = Preconditions.CheckNotNull(systemProperties);
             this.Offset = offset;
             this.EnqueuedTime = enqueuedTime;
             this.DequeuedTime = dequeuedTime;
```

Comment:
This isn't related to your change, but should you validate EnqueuedTime < DequeuedTime?

Reply:
those are unused in our codebase.